jujugui/static/gui/src/app/user/user.js

```diff
@@ -30,6 +41,23 @@ const User = class User {
     this.sessionStorage = cfg.sessionStorage || sessionStorage;
     this.localStorage = cfg.localStorage || localStorage;
     this._external = cfg.externalAuth || null;
+    this.expiration = cfg.expiration || getExpirationDate();
+  }
+
+  get expiration() {
+    return new Date(this.sessionStorage.getItem('expiration'));
+  }
+
+  set expiration(val) {
```

Comment:
Related to `get expiration`: 
This method doesn't do any kind of transformation on `val` so therefore could be anything - probably good to either transform `val` into a specific type, or add docstring as to what `val` should be... Or both?